dhall-lsp-server/src/Dhall/LSP/Backend/Completion.hs

```diff
@@ -158,7 +159,7 @@ completeProjections (CompletionContext context values) expr =
   -- substitute 'dependent lets', necessary for completion of unions
   let values' = toList values
       subs = filter ((/= holeExpr) . snd) $ zip (contextToVariables values') (map snd values')
-      expr' = foldl (\e (x,val) -> subst x val e) expr subs
+      expr' = foldl' (\e (x,val) -> subst x val e) expr subs
```

Comment:
What is `subst` here?

Reply:
Plain substitution: https://hackage.haskell.org/package/dhall-1.27.0/docs/Dhall-Core.html#v:subst